obsreport/obsreport_receiver.go

```diff
@@ -63,7 +83,70 @@ func NewReceiver(cfg ReceiverSettings) *Receiver {
 			tag.Upsert(obsmetrics.TagKeyTransport, cfg.Transport, tag.WithTTL(tag.TTLNoPropagation)),
 		},
 		tracer: cfg.ReceiverCreateSettings.TracerProvider.Tracer(cfg.ReceiverID.String()),
+		meter:  cfg.ReceiverCreateSettings.MeterProvider.Meter(cfg.ReceiverID.String()),
+		logger: cfg.ReceiverCreateSettings.Logger,
+
+		useOtelForMetrics: obsreportconfig.UseOtelForMetrics(),
+		otelAttrs: []attribute.KeyValue{
+			attribute.String(obsmetrics.ReceiverKey, cfg.ReceiverID.String()),
+			attribute.String(obsmetrics.TransportKey, cfg.Transport),
+		},
 	}
+
+	rec.createOtelMetrics()
```

Comment:
should this be behind a check for using otel metrics?

Reply:
Or the check should be at the top of the method.  One of the two.

Reply:
I've decided to add a check at the top.